src/did/did.spec.ts

```diff
@@ -152,6 +161,54 @@ describe('DID Resolver', function test() {
         new RegExp(`^No record found for ${randomIdentity}\\. Is this a valid identity address\\?$`, 'i'),
       );
     });
+
+    it('automatically sets created, proof, and updated properties', async () => {
+      const document = await runtimes[0].did.getDidDocumentTemplate();
+      await runtimes[0].did.setDidDocument(accounts0Did, document);
+      const didDoc = await runtimes[0].did.getDidDocument(accounts0Did);
+      expect(didDoc).to.have.property('created');
+      expect(didDoc).to.have.property('updated');
+      expect(didDoc).to.have.property('proof');
+    });
+
+    it('throws if a DID document has an invalid proof', async () => {
+      const didContract = await (runtimes[0].did as any).getRegistryContract();
+
+      // Create scenario where a document carries a proof that has been
+      //   created over a differing document
+      const documentToSave = await runtimes[0].did.getDidDocumentTemplate();
+      const documentToCreateProofOver = _.cloneDeep(documentToSave);
+      documentToCreateProofOver.updated = new Date(Date.now()).toISOString();
+
+      const invalidProof = await (runtimes[0].did as any).createJwtForDid(
+        documentToCreateProofOver,
+        accounts0Did,
+        DidProofType.EcdsaPublicKeySecp256k1,
+      );
+      documentToSave.proof = {
+        type: DidProofType.EcdsaPublicKeySecp256k1,
+        created: new Date(Date.now()).toISOString(),
+        proofPurpose: 'assertionMethod',
+        verificationMethod: documentToSave.publicKey[0].id,
+        jws: invalidProof,
+      };
+
+      // Set did document with invalid proof manually (the api wouldn't allow this)
+      const identity = await runtimes[0].did.convertDidToIdentity(accounts0Did);
+      const targetHash = await (runtimes[0].did as any).padIdentity(identity);
+      const documentHash = await runtimes[0].dfs.add('did', Buffer.from(JSON.stringify(documentToSave)));
+      await runtimes[0].executor.executeContractTransaction(
+        didContract,
+        'setDidDocument',
+        { from: runtimes[0].activeIdentity },
+        targetHash,
+        documentHash,
+      );
+
+      // Try to retrieve it
+      expect(runtimes[0].did.getDidDocument(accounts0Did))
```

Comment:
```suggestion
      await expect(runtimes[0].did.getDidDocument(accounts0Did))
```

should this be awaited?